### PR TITLE
Don't silently ignore uncaught thread exceptions

### PR DESCRIPTION
**Pull request by obino**

### CODE REVIEW

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/4713/

**Comment by cdonati** [comment]
Do we really want to kill the whole process if an exception is raised while handling a request?

**Comment by obino** [comment]
@cdonati the crash will give us a stack trace and a good way to see it: the AC will get restarted anyway so the effect shouldn't be dramatic at all. I don't see too many downsides to catch cases that shouldn't be common at all anyway.

File: InfrastructureManager/system_manager.py [hunk 1]
```diff
@@ -140,7 +140,11 @@ def get_service_summary(self, secret):
       return self.__generate_response(False,
         InfrastructureManager.REASON_BAD_SECRET)
 
-    monit_stats = subprocess.check_output(["monit", "summary"])
+    try:
+      monit_stats = subprocess.check_output(["monit", "summary"])
+    except CalledProcessError:
+      logging.warn("get_service_summary: failed to query monit.")
+      monit_stats = ""
```
[the remark this answers is not in the document]
Comment (obino) [reply InfrastructureManager/system_manager.py]: what would you suggest? An Error string?

[the remark this answers is not in the document]
Comment (cdonati) [reply InfrastructureManager/system_manager.py]: Since it's a SOAP server, raising an exception and letting the client deal with it should be fine.

**Comment by obino** [comment]
OCD build https://ocd.appscale.com:8080/job/Daily%20Build/5035/

**Pull request merged**